src/PolyhedralGeometry/helpers.jl

```diff
@@ -587,32 +587,39 @@ end
 _find_elem_type(x::Nothing) = Any
 _find_elem_type(x::Any) = typeof(x)
 _find_elem_type(x::Type) = x
-_find_elem_type(x::Polymake.Rational) = QQFieldElem
-_find_elem_type(x::Polymake.Integer) = ZZRingElem
-_find_elem_type(x::AbstractArray) = reshape(_find_elem_type.(x), :)
-_find_elem_type(x::Tuple) = reduce(vcat, _find_elem_type.(x))
-_find_elem_type(x::AbstractArray{<:AbstractArray}) =
-  reduce(vcat, _find_elem_type.(x); init=[])
-_find_elem_type(x::MatElem) = [elem_type(base_ring(x))]
+_find_elem_type(x::Type{Polymake.Rational}) = QQFieldElem
+_find_elem_type(x::Type{Polymake.Integer}) = ZZRingElem
+
+_find_elem_type(x::MatElem) = elem_type(base_ring_type(x))
+_find_elem_type(x::Type{<:AbstractArray}) = _find_elem_type(eltype(x))
+
+function _find_elem_type(x::Union{Tuple,AbstractArray})
+  T = eltype(x)
+  if isconcretetype(T) && T <: RingElem
+    t = _find_elem_type(T)
+    t === Float64 && return Float64
+    return promote_type(t, QQFieldElem)
+  end
 
-function _guess_fieldelem_type(x...)
-  types = filter(!=(Any), _find_elem_type(x))
   T = QQFieldElem
-  for t in types
-    if t == Float64
-      return Float64
-    elseif promote_type(t, T) != T
-      T = t
-    end
+  for y in x
+    t = _find_elem_type(y)
+    t === Any && continue
+    t === Float64 && return Float64
+    T = promote_type(t, T)
```

Comment:
I don't think it will work with using the result of `promote_type`, in many cases this will just give `FieldElem`, e.g.:
```
julia> promote_type(QQFieldElem,QQBarFieldElem)
FieldElem
```
but if that is different from `QQFieldElem` we can assume that we should use the element that we are just checking, i.e., something like `promote_type(t, T) != T && T = t`.

_Edit:_ I am somewhat surprised that it does work now.

Reply:
It looks like we need more tests for this, but it does not work as expected:
```
julia> mm = QQBarFieldElem[1 2 3; 1 3 qb; 4 5 6]
3×3 Matrix{QQBarFieldElem}:
 {a1: 1.00}  {a1: 2.00}  {a1: 3.00}
 {a1: 1.00}  {a1: 3.00}  {a2: 1.30}
 {a1: 4.00}  {a1: 5.00}  {a1: 6.00}

julia> Oscar._guess_fieldelem_type(mm)
FieldElem
```
(on master this returns QQBarFieldElem)